monitor/basic/peer_monitor.go

```diff
@@ -175,36 +188,20 @@ func (mon *Monitor) PublishMetric(ctx context.Context, m *api.Metric) error {
 	return nil
 }
 
-// getPeers gets the current list of peers from the consensus component
-func (mon *Monitor) getPeers(ctx context.Context) ([]peer.ID, error) {
-	ctx, span := trace.StartSpan(ctx, "monitor/basic/getPeers")
-	defer span.End()
-
-	var peers []peer.ID
-	err := mon.rpcClient.CallContext(
-		ctx,
-		"",
-		"Cluster",
-		"ConsensusPeers",
-		struct{}{},
-		&peers,
-	)
-	if err != nil {
-		logger.Error(err)
-	}
-	return peers, err
-}
-
 // LatestMetrics returns last known VALID metrics of a given type. A metric
 // is only valid if it has not expired and belongs to a current cluster peers.
 func (mon *Monitor) LatestMetrics(ctx context.Context, name string) []*api.Metric {
 	ctx, span := trace.StartSpan(ctx, "monitor/basic/LatestMetrics")
 	defer span.End()
 
-	latest := mon.metrics.Latest(name)
+	latest := mon.metrics.LatestValid(name)
+
+	if mon.peers == nil {
+		return latest
+	}
 
 	// Make sure we only return metrics in the current peerset
-	peers, err := mon.getPeers(ctx)
+	peers, err := mon.peers(ctx)
 	if err != nil {
 		return []*api.Metric{}
```

Comment:
return nil

Reply:
@lanzafame in the majority we SHOULD return the empty slice and not nil: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices . nil slices encode to `null` in json, while empty slices encode to `[]`. These responses are used to encode JSON and no metrics should not be `null`, but rather `[]` .